swap_meet/vendor.py

```diff
@@ -1,2 +1,146 @@
+        # no items in the inventory that match the category
+        # argument the method returns an empty list
+        item_in_category = []
+        # check vendor inventory by item
+        for item in self.inventory:
+            #if item in inventory 
+            if item.category == category:
+                # return item in in a list
+                item_in_category.append(item)
+        return item_in_category
+
+
+    # method takes argument category str
+    def get_best_by_category(self, category=""):
+            
+            # if there is no items in inventory return None
+            best_item = None
+            # look thru inventory for item 
+            for item in self.inventory:
+                # matching category
+                if item.category == category:
+                    # if there is no items in inventory return None
+                    #it returns a single item even if there are dupes
+                    if best_item is None or item.condition > best_item.condition:
+                        best_item = item
+            # returns this item with highest conditin and matching category
+            return best_item
+            
+
+    def swap_best_by_category(self,  other_vendor,my_priority,their_priority):
+        # my_best_item = self.get_best_by_category(their_priority)
+        # their_best_item = self.get_best_by_category(my_priority)
+        # think this would be part of a solution to the asserts i am not
+        # passing but couldn't get it to work before due date will try later
+        # as im trying to swap best item by category
+        my_best_item = None
+        their_best_item = None
+        # check best item in my inventory that matches their_priority category
+        for item in self.inventory:
+            if item.category == their_priority:
+            # If the Vendor has no item that matches their_priority category, it returns False
+                if my_best_item is None:
+                    my_best_item = item
+            #  check best item in other_vendor's inventory that matches my_priority
+            for item in other_vendor.inventory:
+                if item.category == my_priority:
+                # If other_vendor has no item that matches my_priority category, it returns False
+                    if their_best_item is None:
+                        their_best_item = item
+
+                # if it is in inventory swap items
+                if my_best_item is not None and their_best_item is not None:
+                    # self.inventory.remove(my_best_item)
+                    # other_vendor.inventory.append(my_best_item)
+                    # other_vendor.inventory.remove(their_best_item)
+                    # self.inventory.append(their_best_item)
+                    # my attempt to drying code with swap_items
+                    self.swap_items(other_vendor, my_best_item, their_best_item)
+                    return True
```

Comment:
As you mentioned, the logic in this function doesn't compare condition, and it's missing the right part of using `get_best_by_category`. To move forward on this, I would recommend thinking "outside" the `for item in self.inventory`.

I've personally wrote a solution that works with your code; let me know if you want to check or compare or have questions!